service/client.ts

```diff
@@ -23,6 +24,42 @@ import isGrpcError from "utils/isGrpcError";
 
 const URL = process.env.NEXT_PUBLIC_API_BASE_URL;
 
+async function http<T>(endpoint: string, configOverride: RequestInit): Promise<T> {
+  const url = path.join(URL, endpoint)
+  const config = {
+    headers: {
+      "content-type": "application/json"
+    },
+    ...configOverride
+  }
+  const request = new Request(url, config)
+  const response = await fetch(request)
+
+  // if(!response.ok) {
+  //   throw new Error(response.statusText)
+  // }
+
+  // may error if there is no body, return empty array
+  return response.json().catch(() => ({}))
+}
+
+export async function get<T>(path: string, config?: RequestInit): Promise<T> {
+  const init = {method: 'get', ...config}
+  return await http<T>(path, init)
+}
+
+export async function post<T, U>(path: string, body: T, config?: RequestInit): Promise<U> {
+  const init = {method: 'post', body: JSON.stringify(body), ...config}
+  return await http<U>(path, init)
+}
+
+export async function put<T, U>(path: string, body: T, config?: RequestInit): Promise<U> {
+  const init = {method: 'put', body: JSON.stringify(body), ...config}
+  return await http<U>(path, init)
+}
```

Comment:
I think we should have the new client code in its own file, so that when it comes to the time of deletion, we can just delete this file altogether. The `index.ts` in this folder should be used as the public interface to expose these new methods as a lot of our tests import through that, and the mocking requires that to work properly.

Reply:
moved to `http.ts`